src/main.rs

```diff
@@ -57,10 +58,23 @@ impl<'a> Generator<'a> {
         handlebars.register_templates_directory(".hbs", "templates")?;
         handlebars.register_helper("month_name", Box::new(hb_month_name_helper));
 
+        let file_url = format!(
+            "file:///{}/",
```

Comment:
I don't believe the `file://` protocol works on Windows. Should we prefix only when `#[cfg(not(target_os = "windows"))]`?

Reply:
I didn't test it on Windows, but I did look it up and concluded it should work. Did you test it?

Reply:
https://en.wikipedia.org/wiki/File_URI_scheme#Windows_2

Reply:
Browsers do allow the `file://` protocol on windows.

Reply:
You're right, it's not the `file::///` protocol, but rather that `canonicalize` uses UNC (universal naming convention) file paths (i.e., the `\\?\`). I don't know enough about Windows file systems or `canonicalize` to know what the right thing to do here is. Is it fine to just remove any `\\?\` prefix from what is returned from `canonicalize`?   

FYI: This is what gets printed on my Windows machine: "file:///\\?\C:\Users\ryanl\Code\blog.rust-lang.org\site/2022\05\19\Rust-1.61.0.html"

Reply:
Does it work now? :)

(Not the most robust code, but it probably works in all the common cases. ^^')

Reply:
So, this works better, but still breaks in the terminal.

The url is rendered as "file:///C:/Users/ryanl/Code/blog.rust-lang.org/site/inside-rust\index.html". Notice the mixed path separators. 

Notice, the out-directory path works fine, and I can directly click on it to open my browser. Unfortunately, it can't handle the mixed used of path separators though, so it doesn't open the full path. When you render the path to specific file can you do the call to `.replace(std::path::MAIN_SEPARATOR, "/")`?

Reply:
~~`.replace('\\', "[pleaseswitchtoanoperatingsystemwithjustonepathseparator😢]")`~~

Reply:
Works!